### ISSUE TITLE
Use bindgen to obtain derived constants that we use from libdevmapper.h

### ISSUE THREAD

**Issue by mulkieran** [stratis-storage/devicemapper-rs#730]
This includes all the DmCookie constants and also the 16 bit left shift in ```to_ioctl_hdr()``` which is really ```DM_UDEV_FLAGS_SHIFT```.

**Comment by mulkieran** [comment]
@bmr-cymru I think it'll probably be easy and appropriate for you to start using these constants in the approved manner as part of your ongoing semaphore work, so I'm assigning this one to you.

**Comment by bmr-cymru** [comment]
The DM udev flags (and the shift/mask constants) are all defined in libdevmapper.h so we will need to extend bindgen to selectively include content from this header as well. I have seen a few references in the docs on ways to accomplish that and I'll try to get that working first while I continue to work on generalising the udev sync handling.

One quirk that will have to be dealt with is that the device-mapper headers use `await` as an identifier (since it is also a Rust keyword):

```
   Compiling devicemapper-sys v0.1.2 (/home/breeves/src/git/devicemapper-rs/devicemapper-rs-sys)
error: expected identifier, found keyword `await`
    --> /home/breeves/src/git/devicemapper-rs/target/debug/build/devicemapper-sys-ac67731d8059b72d/out/bindings.rs:9586:9
     |
9586 |         await: *mut f64,
     |         ^^^^^ expected identifier, found keyword
     |
help: escape `await` to use it as an identifier
     |
9586 |         r#await: *mut f64,
     |         ++

error: expected identifier, found keyword `await`
    --> /home/breeves/src/git/devicemapper-rs/target/debug/build/devicemapper-sys-ac67731d8059b72d/out/bindings.rs:9594:9
     |
9594 |         await: *mut f64,
     |         ^^^^^ expected identifier, found keyword
     |
help: escape `await` to use it as an identifier
     |
9594 |         r#await: *mut f64,
     |         ++

error: expected identifier, found keyword `await`
    --> /home/breeves/src/git/devicemapper-rs/target/debug/build/devicemapper-sys-ac67731d8059b72d/out/bindings.rs:9602:9
     |
9602 |         await: *mut f64,
     |         ^^^^^ expected identifier, found keyword
     |
help: escape `await` to use it as an identifier
     |
9602 |         r#await: *mut f64,
     |         ++

error: could not compile `devicemapper-sys` due to 3 previous errors

```

**Comment by mulkieran** [comment]
> The DM udev flags (and the shift/mask constants) are all defined in libdevmapper.h so we will need to extend bindgen to selectively include content from this header as well. I have seen a few references in the docs on ways to accomplish that and I'll try to get that working first while I continue to work on generalising the udev sync handling.
> 
> One quirk that will have to be dealt with is that the device-mapper headers use `await` as an identifier (since it is also a Rust keyword):
> 
> ```
>    Compiling devicemapper-sys v0.1.2 (/home/breeves/src/git/devicemapper-rs/devicemapper-rs-sys)
> error: expected identifier, found keyword `await`
>     --> /home/breeves/src/git/devicemapper-rs/target/debug/build/devicemapper-sys-ac67731d8059b72d/out/bindings.rs:9586:9
>      |
> 9586 |         await: *mut f64,
>      |         ^^^^^ expected identifier, found keyword
>      |
> help: escape `await` to use it as an identifier
>      |
> 9586 |         r#await: *mut f64,
>      |         ++
> 
> error: expected identifier, found keyword `await`
>     --> /home/breeves/src/git/devicemapper-rs/target/debug/build/devicemapper-sys-ac67731d8059b72d/out/bindings.rs:9594:9
>      |
> 9594 |         await: *mut f64,
>      |         ^^^^^ expected identifier, found keyword
>      |
> help: escape `await` to use it as an identifier
>      |
> 9594 |         r#await: *mut f64,
>      |         ++
> 
> error: expected identifier, found keyword `await`
>     --> /home/breeves/src/git/devicemapper-rs/target/debug/build/devicemapper-sys-ac67731d8059b72d/out/bindings.rs:9602:9
>      |
> 9602 |         await: *mut f64,
>      |         ^^^^^ expected identifier, found keyword
>      |
> help: escape `await` to use it as an identifier
>      |
> 9602 |         r#await: *mut f64,
>      |         ++
> 
> error: could not compile `devicemapper-sys` due to 3 previous errors
> ```

I think  you should file a PR against rust-bindgen for that. They have a list of things to mangle[1] in the source, and ```await``` is not in it. But then we should defer that part until the next version of bindgen comes out.

[1] https://github.com/rust-lang/rust-bindgen/blob/master/src/ir/context.rs#L824

**Comment by bmr-cymru** [comment]
I was able to work around this for now by using allowlists to control what symbols bindgen includes - since we only need the constants it filters down the bindings.rs file considerable.

I've also filed a pull request with rust-lang/rust-bindgen#2247 to add the missing "await" string to the mangling list.

**Comment by mulkieran** [comment]
The PR has been merged, and should be in the release after 0.60.1.
